### PR TITLE
checker fixes: None, trailing ws, list init

### PR DESCRIPTION
**Pull request by mwichmann**
checker-suggested fixes:

- Mostly, fix remaining instances of comparing none without "is"
- Some trailing whitespace on lines
- A couple of instances of list init followed immediately by several appends, turned into a single list init
- Some double comparisons turned into a single expression

Functionally, these should be a no-op, so no doc impacts.

Signed-off-by: Mats Wichmann <mats@linux.com>

## Contributor Checklist:

* [X] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [X] I have updated `master/src/CHANGES.txt` directory (and read the `README.txt` in that directory)
* [X] I have updated the appropriate documentation

### CODE REVIEW

**Comment by mwichmann** [comment]
Huh. Sider complaint is only on lines that changed, so in `test_get_env_var` lines 450-453 are not commented on (unchanged), while lines 454-458 (changed) got the complaint.  The complaint is `'get_environment_var' may be undefined, or defined from star imports: SCons.Util` which is true - it *does* come from a star import.  Why is this an issue?  Can I just leave this as is?

**Comment by mwichmann** [comment]
Grr, looks like I hit the wrong button... it closed it, that was not intended.

File: src/engine/SCons/SConf.py [hunk 1]
```diff
@@ -1067,7 +1067,7 @@ def CheckLibWithHeader(context, libs, header, language,
     """
     prog_prefix, dummy = \
                  createIncludesFromHeaders(header, 0)
-    if libs == []:
+    if not libs:
```
Comment (bdbaddog) [inline src/engine/SCons/SConf.py:1070]: Not sure it matters, but this would be true of libs = False or 0 or other values and not just for []?

**Comment by bdbaddog** [comment]
> Huh. Sider complaint is only on lines that changed, so in `test_get_env_var` lines 450-453 are not commented on (unchanged), while lines 454-458 (changed) got the complaint. The complaint is `'get_environment_var' may be undefined, or defined from star imports: SCons.Util` which is true - it _does_ come from a star import. Why is this an issue? Can I just leave this as is?

My guess is you touched that code and so it applies it's rules.
I'll merge.

**Comment by mwichmann** [comment]
Indeed it only reports on touched lines, it just makes it odd in that case :)

**Pull request merged**